salt/states/dvs.py

```diff
@@ -0,0 +1,700 @@
+                    changes_string = ''
+                    for p in props_to_updated_values:
+                        if p == 'infrastructure_traffic_resource_pools':
+                            changes_string += \
+                                    '\tinfrastructure_traffic_resource_pools:\n'
+                            for idx in range(len(props_to_updated_values[p])):
+                                d = props_to_updated_values[p][idx]
+                                s = props_to_original_values[p][idx]
+                                changes_string += \
+                                        ('\t\t{0} from \'{1}\' to \'{2}\'\n'
+                                         ''.format(d['key'], s, d))
+                        else:
+                            changes_string += \
+                                    ('\t{0} from \'{1}\' to \'{2}\'\n'
+                                     ''.format(p, props_to_original_values[p],
+                                               props_to_updated_values[p]))
+                    comments.append(
+                        'State dvs_configured will update DVS \'{0}\' '
+                        'in datacenter \'{1}\':\n{2}'
+                        ''.format(dvs_name, datacenter_name, changes_string))
+                    log.info(comments[-1])
+                else:
+                    __salt__['vsphere.update_dvs'](
+                        dvs_dict=props_to_updated_values,
+                        dvs=dvs_name,
+                        service_instance=si)
+                    comments.append('Updated DVS \'{0}\' in datacenter \'{1}\''
+                                    ''.format(dvs_name, datacenter_name))
+                    log.info(comments[-1])
+                changes.update({'dvs': {'new': props_to_updated_values,
+                                        'old': props_to_original_values}})
+        __salt__['vsphere.disconnect'](si)
+    except salt.exceptions.CommandExecutionError as exc:
+        log.error('Error: {0}\n{1}'.format(exc, traceback.format_exc()))
+        if si:
+            __salt__['vsphere.disconnect'](si)
+        if not __opts__['test']:
+            ret['result'] = False
+        ret.update({'comment': str(exc),
+                    'result': False if not __opts__['test'] else None})
+        return ret
+    if not comments:
+        # We have no changes
+        ret.update({'comment': ('DVS \'{0}\' in datacenter \'{1}\' is '
+                                'correctly configured. Nothing to be done.'
+                                ''.format(dvs_name, datacenter_name)),
+                    'result': True})
+    else:
+        ret.update({'comment': '\n'.join(comments)})
+        if __opts__['test']:
+            ret.update({'pchanges': changes,
+                        'result': None})
+        else:
+            ret.update({'changes': changes,
+                        'result': True})
+    return ret
+
+
+def _get_diff_dict(dict1, dict2):
```

Comment:
Would it be better to move these private dictionary difference functions over to `salt.utils.dictdiffer`?

Reply:
Yep, I already created a `RecursiveDictDiffer` in `salt.utils.dictdiffer` which should be used here (it's in a different pull request). This is an pretty old code however and I didn't have time to update it. It's not such a trivial change. That's why I've upstreamed this code and will revisit later.